index.js

```diff
@@ -21,7 +21,7 @@ const hook = createHook({
 
     asyncResources.set(asyncId, {
       type,
-      resource,
+      resourceRef: new WeakRef(resource),
```

Comment:
Just to make sure, we need both the `WeakRef` and your implementation from https://github.com/mafintosh/why-is-node-running/issues/84#issuecomment-2555433780 to make this work properly? Using only your original implantation in the linked comment won't work?

Reply:
My assumption is yes, no one should be holding strong reference to resource. But I haven't verified it. Will check and confirm.

Reply:
We could possibly not store the resource at all, I don't think we actually refer to it anywhere explicitly.

Reply:
It is used a little bit to check if resource is unrefed. I think it is related to unref method of timers: https://github.com/mafintosh/why-is-node-running/pull/52

Reply:
Ok, let me know if this will also work without `WeakRef`, if it does we can use your original implementation, if not then we can use this current incarnation.

Reply:
I've checked and can confirm that we need to store resource wrapped into `WeakRef`

Reply:
Thanks for checking!